### PR TITLE
lychee: 0.5.0 -> 0.7.0

### PR DESCRIPTION
**Pull request by tuxinaut**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/lychee/default.nix [hunk 1]
```diff
@@ -7,16 +7,16 @@
...
   pname = "lychee";
-  version = "0.5.0";
+  version = "0.7.0";
 
   src = fetchFromGitHub {
     owner = "lycheeverse";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "03dsp0384mwr51dkqfl25xba0m17sppabiz7slhxcig89b0ksykm";
+    rev = "f7f9485be0dc9435cee302cc49bb7f8a705fdff8";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/lychee/default.nix:15]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/networking/lychee/default.nix [hunk 2]
```diff
@@ -7,16 +7,16 @@
...
+    sha256 = "0kpwpbv0dqb0p4bxjlcjas6x1n91rdsvy2psrc1nyr1sh6gb1q5j";
   };
 
-  cargoSha256 = "08y2wpm2qgm2jsy257b2p2anxy4q3bj2kfdr5cnb6wnaz9g4ypq2";
+  cargoSha256 = "1b915zkg41n3azk4hhg6fgc83n7iq8p7drvdyil2m2a4qdjvp9r3";
 
   nativeBuildInputs = [ pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/lychee/default.nix:22]: Do tests now pass?

Comment (tuxinaut) [reply pkgs/tools/networking/lychee/default.nix:22]: Nope :disappointed:

File: pkgs/tools/networking/lychee/default.nix (hunk 1 above)
Comment (tuxinaut) [reply pkgs/tools/networking/lychee/default.nix:15]: Sure, I would prefer that as well, but the suggested code produces this.

```
copying path '/nix/store/bh3psvj27bp898kxmgbn8skxb7sx9508-git-2.31.1' from 'https://cache.nixos.org'...

trying https://github.com/lycheeverse/lychee/archive/v0.7.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   126  100   126    0     0    402      0 --:--:-- --:--:-- --:--:--   402
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download source from any mirror
builder for '/nix/store/m4c3zdc8p4wzi5z2nankmh1i6nd4kdfp-source.drv' failed with exit code 1
cannot build derivation '/nix/store/chhjhx5gfspmi5q934hq2nif7axmlikq-lychee-0.7.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/chhjhx5gfspmi5q934hq2nif7axmlikq-lychee-0.7.0.drv' failed
```

Comment (SuperSandro2000) [reply pkgs/tools/networking/lychee/default.nix:15]:
```suggestion
    rev = version;
```
its actually

Comment (tuxinaut) [reply pkgs/tools/networking/lychee/default.nix]: :+1: thanks for the hint.

File: pkgs/tools/networking/lychee/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/networking/lychee/default.nix:22]: You can try to contact upstream and figure out why the tests fail.

Comment (tuxinaut) [reply pkgs/tools/networking/lychee/default.nix:22]: I will give it a try :+1:

**Pull request merged**